sqlmesh/core/snapshot/evaluator.py

```diff
@@ -2239,9 +2362,17 @@ def create(
                 model,
                 is_table_deployable,
                 render_kwargs,
+                skip_grants,
                 **kwargs,
             )
 
+        if not skip_grants:
+            # Apply grants after SCD Type 2 table creation
```

Comment:
Don't we recreate a table for SCD 2 every time? Shouldn't we add grants on insert then too?

Reply:
Added _apply_grants to SCD2 insert: https://github.com/TobikoData/sqlmesh/pull/5275/commits/90a6818e44774a0aa80b0ab7806277c61ca3ec45